Automattic/wp-calypso
# Plans: Make storage selection calculations in plan prices optional

Related to https://github.com/Automattic/martech/issues/3403 and https://github.com/Automattic/wp-calypso/pull/96357#pullrequestreview-2441944966

## Proposed Changes

* Disable storage add-on selection updates in header price for the `calypso_plans_page_emphasize_longer_plan_savings` experiment with a new `reflectStorageSelectionInPlanPrices` flag
* Pass `reflectStorageSelectionInPlanPrices` to context to reference throughout `plans-grid-next`. If `false`, prevent storage add-on calculations in features grid + comparison grid plan pricing.

## Why are these changes being made?


* See p1726619377025979/1724962428.022989-slack-C07ABDFQEMS

## Testing Instructions



* Go to /start/plans?flags=plans/term-savings-price-display
* Ensure that when storage add-ons are selected for business or ecommerce annual plans, the header price doesn't change

## Pre-merge Checklist
